code/game/objects/items/weapons/storage/storage_base.dm

```diff
@@ -49,6 +51,7 @@
 	. = ..()
 	can_hold = typecacheof(can_hold)
 	cant_hold = typecacheof(cant_hold)
+	cant_hold_override = typecacheof(cant_hold_override)
```

Comment:
```suggestion
	cant_hold = typecacheof(cant_hold) - typesof(cant_hold_override)
```
This will result in the `cant_hold` list to not have the typesof the `cant_hold_override` list. Thus you can remove the second `is_type_in_typecache` check